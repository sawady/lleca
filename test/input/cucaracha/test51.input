
fun main() {
    v := [1, 2, 3]
    putNum(#v)
    putChar(10)
}
