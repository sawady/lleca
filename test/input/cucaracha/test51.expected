Cons(
  Function(
    main,
    Nil,
    Unit,
    Cons(
      StmtAssign(
        v,
        ExprVecMake(
          Cons(
            ExprConstNum(
              1
            ),
            Cons(
              ExprConstNum(
                2
              ),
              Cons(
                ExprConstNum(
                  3
                ),
                Nil
              )
            )
          )
        )
      ),
      Cons(
        StmtCall(
          putNum,
          Cons(
            ExprVecLength(
              v
            ),
            Nil
          )
        ),
        Cons(
          StmtCall(
            putChar,
            Cons(
              ExprConstNum(
                10
              ),
              Nil
            )
          ),
          Nil
        )
      )
    )
  ),
  Nil
)
